From 742d0308e8347311bb5f5b847f8716a35a9907aa Mon Sep 17 00:00:00 2001
From: Felix Manlunas <fmanlunas@marvell.com>
Date: Mon, 21 Oct 2019 10:58:01 -0700
Subject: [PATCH 0304/1921] octeontx2-af: Increase mbox response timeout to 2
 seconds

It used to be that the intended 1 second mbox response timeout was not very
precise such that the effective timeout was longer than 1 second.  Recently
code was checked in to make the timeout very precise.

It turns out that the old, imprecise, longer-than-1-second timeout was
beneficial.  Now that it's been replaced with a precise 1-second timeout,
we sometimes see these alarming error messages from ethtool:

    Cannot set new settings: Input/output error
    Cannot set private flags: Input/output error

These turn out to be false alarms because the operations (of changing
Ethernet speed or setting private flags) actually take effect.

One second is not enough time for ATF to detect the arrival of a CGX_CMD,
to execute the command, and to send back a response.

Increase the timeout to 2 seconds for ample safety margin.

Change-Id: Ia663fdf55a7a1417fc2d8b9ee751cab0903a7710
Signed-off-by: Felix Manlunas <fmanlunas@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/17678
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
(cherry picked from commit b842f3df534a85e70811b4490900a10d17f46f99)
Reviewed-on: https://sj1git1.cavium.com/17824
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/mbox.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/mbox.h b/drivers/net/ethernet/marvell/octeontx2/af/mbox.h
index d568fb07fc51..174cb52b67ea 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/mbox.h
+++ b/drivers/net/ethernet/marvell/octeontx2/af/mbox.h
@@ -36,7 +36,7 @@
 
 #define INTR_MASK(pfvfs) ((pfvfs < 64) ? (BIT_ULL(pfvfs) - 1) : (~0ull))
 
-#define MBOX_RSP_TIMEOUT	1000 /* Time(ms) to wait for mbox response */
+#define MBOX_RSP_TIMEOUT	2000 /* Time(ms) to wait for mbox response */
 
 #define MBOX_MSG_ALIGN		16  /* Align mbox msg start to 16bytes */
 
-- 
2.31.1

